tests/test_linestring.py

```diff
@@ -6,6 +6,109 @@
 from shapely.geometry import LineString, asLineString, Point, LinearRing
 
 
+def test_from_coordinate_sequence():
+    # From coordinate tuples
+    line = LineString(((1.0, 2.0), (3.0, 4.0)))
+    assert len(line.coords) == 2
+    assert line.coords[:] == [(1.0, 2.0), (3.0, 4.0)]
+
+    line = LineString([(1.0, 2.0), (3.0, 4.0)])
+    assert line.coords[:] == [(1.0, 2.0), (3.0, 4.0)]
+
+
+def test_from_coordinate_sequence_3D():
+    line = LineString(((1.0, 2.0, 3.0), (3.0, 4.0, 5.0)))
+    assert line.has_z
+    assert line.coords[:] == [(1.0, 2.0, 3.0), (3.0, 4.0, 5.0)]
+
+
+def test_from_points():
+    # From Points
+    line = LineString((Point(1.0, 2.0), Point(3.0, 4.0)))
+    assert line.coords[:] == [(1.0, 2.0), (3.0, 4.0)]
+
+    line = LineString([Point(1.0, 2.0), Point(3.0, 4.0)])
+    assert line.coords[:] == [(1.0, 2.0), (3.0, 4.0)]
+
+
+def test_from_mix():
+    # From mix of tuples and Points
+    line = LineString((Point(1.0, 2.0), (2.0, 3.0), Point(3.0, 4.0)))
+    assert line.coords[:] == [(1.0, 2.0), (2.0, 3.0), (3.0, 4.0)]
+
+
+def test_from_linestring():
+    # From another linestring
+    line = LineString(((1.0, 2.0), (3.0, 4.0)))
+    copy = LineString(line)
+    assert copy.coords[:] == [(1.0, 2.0), (3.0, 4.0)]
+    assert lgeos.GEOSGeomType(copy._geom).decode('ascii') == 'LineString'
```

Comment:
perhaps it's also time to move away from ctypes functions and use:
```suggestion
    assert copy.geom_type == 'LineString'
```
(similar applies to below)

Reply:
Yes! I was assuming this was done for a reason (eg because each of the subclasses simply hardcodes its geometry type instead of calling into GEOS), but the `geom_type` attribute actually calls `GEOSGeomTypeId`, so will indeed update.